java/src/main/native/src/jni_writer_data_sink.hpp

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "cudf_jni_apis.hpp"
+#include "jni_utils.hpp"
+
+#include <cudf/io/data_sink.hpp>
+
+namespace cudf::jni {
+
+constexpr long MINIMUM_WRITE_BUFFER_SIZE = 10 * 1024 * 1024; // 10 MB
+
+class jni_writer_data_sink final : public cudf::io::data_sink {
```

Comment:
The class name doesn't need to have `jni_` prefix.

Reply:
....because it is already in `cudf::jni::` namespace.

Reply:
`jni_writer_data_sink` is a preexisting class. It's just been moved here. I'd rather we didn't change its name.
(Removing `jni_` would make this the `writer_data_sink`, which doesn't seem very specific. This is the `data_sink` that's used from the JNI layer.)

Reply:
I'll see about renaming this class in #12456.

Reply:
#12458 addresses the rename.